float64 vbatt
float64 ibatt

float64 e_batt
float64 p_batt

float64 reg_energy
